fix: kwargs.pop("redis_url") KeyError: 'redis_url'

jdwuarin opened the pull request:
This occurred when redis_url was not passed as a parameter even though a REDIS_URL env variable was present.
This occurred for all methods that eventually called any of: (from_texts, drop_index, from_existing_index) - i.e. virtually all methods in the class. 
This fixes it

dev2049 commented on langchain/vectorstores/redis.py:
```diff
@@ -273,7 +273,10 @@ def from_texts(
             # We need to first remove redis_url from kwargs,
             # otherwise passing it to Redis will result in an error.
-            kwargs.pop("redis_url")
+            try:
```
instead of try/except could we explicitly check
```python
if "redis_url" in kwargs:
  del kwargs["redis_url"]
```

jdwuarin replied:
Yes, that makes sense. I've updated to using an if statement. I've kept the original `pop` syntax though rather than moving to `del` as it seemed to work fine as it was.

jdwuarin committed:
fix: kwargs.pop("redis_url")
KeyError: 'redis_url'

this occurred when redis_url was not passed to the editted methods (from_texts, drop_index, from_existing_index)
even though a REDIS_URL env variable was present. This fixes it

hwchase17 reviewed (approved):
thanks!
